activities/ActivityTemplate/VueJS/js/components/SugarIcon.js

```diff
@@ -0,0 +1,47 @@
+Vue.component('sugar-icon', {
+	data: function () {
+		return {
+			icon: null
+		}
+	},
+	mounted() {
+		var vm = this;
+		requirejs(["sugar-web/graphics/icon"], function (icon) {
+			vm.icon = icon;
+		});
+	},
+	methods: {
+		colorize: function (element, colors, callback) {
+			this.icon.colorize(element, colors, callback);
+		},
+
+		generateIconWithColors: function (path, colors, callback) {
```

Comment:
Not a big fan of the idea to use callback function here. Other components rely on events. May be it's time here to use promise?

Reply:
Yes, we can use a Promise here. Events are useful when we don't need to have a flow after function call and a different method can handle the event (Eg. Acceleration updates).